cloudcix_primitives/cpu_lxd.py

```diff
@@ -1,13 +1,71 @@
+'''
+CPU limit update for LXD instances.
+'''
 # stdlib
 from typing import Tuple
-# lib
+# libs
+from cloudcix.rcc import API_SUCCESS, CHANNEL_SUCCESS, comms_lxd
 # local
+from cloudcix_primitives.utils import HostErrorFormatter, LXDCommsWrapper
 
-
-all__ = [
+__all__ = [
     'update',
 ]
 
+def update(
+        endpoint_url: str,
+        project: str,
+        container_name: str,
+        cpu: int,
+        verify_lxd_certs: bool = True
+) -> Tuple[bool, str]:
+    
+    """ Update the CPU limit of an LXD instance.
+    :param endpoint_url: The endpoint URL for the LXD Host.
+    :param project: The LXD project name.
+    :param instance_name: The name of the LXD instance.
+    :param cpu: The number of CPU cores to set.
+    :param verify_lxd_certs: Boolean to verify LXD certs.
+    :return: A tuple with a boolean flag indicating success or failure, a message, and a dictionary of successful payloads.
+    """
+    # Define message
+    messages = {
+        1000: f'Successfully updated the CPU limit for containers {container_name} on {endpoint_url}',
+        3021: f'Failed to connect to {endpoint_url} for containers.get payload',
+        3022: f'Failed to run containers.get payload on {endpoint_url}. Payload exited with status ',
+        3023: f'Failed to set CPU limit for containers {container_name}. Error: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+        
+        # Get the instance
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+
+        # Update the CPU limit
+        try:
+            instance.config['limits.cpu'] = str(cpu)
+            instance.save(wait=True)
+            fmt.add_successful('containers.set', {'limits.cpu': str(cpu)})
+        except Exception as e:
+            return False, f"{prefix+3}: {messages[prefix+3]}: {e}", fmt.successful_payloads
+
+        return True, '', fmt.successful_payloads
+    
+    status, msg = run_host(endpoint_url, 3020, {})
```

Comment:
```suggestion
    status, msg, successful_payloads = run_host(endpoint_url, 3020, {})
```

Resolve the below encountered when running tests
```
Traceback (most recent call last):
  File "test_cpu_lxd.py", line 40, in <module>
    status, msg, a  = cpu_lxd.update(
  File "/lib/python3.8/site-packages/cloudcix_primitives/cpu_lxd.py", line 66, in update
    status, msg = run_host(endpoint_url, 3020, {})
ValueError: too many values to unpack (expected 2)
```